### DISCUSSION TITLE
[Tiddler file naming] Save plugin.info tiddler in specified path with .info extension

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#5249] [Q&A] [answered]
I use  `$:/config/FileSystemPaths` for customizing tiddler file naming. I have a `plugin.info` tiddler and I like to

1. save it in `system/plugins/kookma/tamasha`
2. save on the disk with `.info` extension NOT  `.tid`

I used the below filter

- $:/config/FileSystemPaths
`[title[plugin.info]addprefix[system/plugins/kookma/tamasha]]`

but no success! Would you please advise me on this.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome 87
 - Node.JS 12 .19.0
 - Version Tiddlywiki 5.1.23 on Node.JS

**Comment by Jermolene** [comment]
Hi @kookma

> 2. save on the disk with `.info` extension NOT  `.tid`

the trouble is on reading the file back the system would have no way of knowing that the `.info` file was to be treated as a `.tid` file.

**Comment by kookma** [comment]
@Jermolene  - Thank you for clarification.  I hope some day I can solve this problem -:)

**Comment by joshuafontany** [comment, marked as answer]
With 5.1.23 you can use a `$:/config/FileSystemExtensions` like this:

[suffix[.info]then[.info]]

This will a) force a *.meta file to be written to disk holding all other field but the text field, and b) the text field will be written to disk as the "safe-title"plus the custom-extention-from-filter.

This is handled separately from the paths defined in $:/config/FileSystemPaths. You can remove the ".info" from the tiddler's title in the $:/config/FileSystemPaths filters.

**Reply by kookma** [reply]
@joshuafontany - thank you. That means I can write the `my-tiddler.info` as file `my-tiddler.info` on the disk. So with 5.1.23 I can implement what I asked! Am I right?

**Comment by kookma** [comment]
@joshuafontany - I was able to write to disk and read from disk using your suggestion

- `$:/config/FileSystemExtensions` contains
`[suffix[.info]then[.info]]`

and 

- `$:/config/FileSystemPaths` contains

```
[suffix[.info]!has[draft.of]removesuffix[.info]removeprefix[$:/]addprefix[system/]]

[is[system]!has[draft.of]removeprefix[$:/]addprefix[system/]]

[is[draft]search-replace:g:regexp[/|\\],[_]addprefix[drafts/]]
```

So, it works great! I can create plugin.info tiddler of type JSON and write/read to disk as `plugin.info`. Thank you

**caveat**
The strange behavior is, if I change  line one with line three in `$:/config/FileSystemPaths` Tiddlywiki save plugin.info as `plugin.info.info`.  I cannot understand this. May be it is because of first line and third line both create a name for plugin.info tiddler and first takes priority.

**Reply by saqimtiaz** [reply]
@kookma in both config tiddlers, the first path to match wins. So the most specific filters need to be higher up.

Documentation: https://tiddlywiki.com/prerelease/#Customising%20Tiddler%20File%20Naming

**Reply by kookma** [reply]
Thank you Saq! So here the behavior is different from say `<$lisy filter="filter-run1 filter-run2">`.  I bookmarked the doc page.

**Reply by saqimtiaz** [reply]
@kookma consider the use case: if multiple filters match to modify a path, we cannot apply them all and still get a coherent path. Therefore we use the first filter that matches.

**Reply by kookma** [reply]
I got it! Thank you
